stl/inc/algorithm

```diff
@@ -4096,14 +4096,14 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
-#ifdef __cpp_lib_concepts
 template <class _InIt, class _OutIt>
-concept _Can_reread_dest = forward_iterator<_OutIt> && same_as<iter_value_t<_InIt>, iter_value_t<_OutIt>>;
-#else // ^^^ defined(__cpp_lib_concepts) / !defined(__cpp_lib_concepts) vvv
-template <class _InIt, class _OutIt>
-_INLINE_VAR constexpr bool _Can_reread_dest = _Is_fwd_iter_v<_OutIt> //
+#if defined(__cpp_lib_concepts)
+concept _Can_reread_dest
+#else
+_INLINE_VAR constexpr bool _Can_reread_dest
+#endif
+    = _Is_fwd_iter_v<_OutIt> //
     && is_same_v<_Iter_value_t<_InIt>, _Iter_value_t<_OutIt>>;
-#endif // __cpp_lib_concepts

```

Comment:
Can you explain what the drive-by fix is? I see `forward_iterator` being replaced with the "stronger" (?) `_Is_fwd_iter_v` for concepts mode.

Reply:
I think this is not correct. This should be `_Is_ranges_fwd_iter_v`

Reply:
It was a broken test before; `_Can_reread_dest` is used on a mutable iterator, and so it should use the Cpp17*Iterator type.

Reply:
Yes, this is fixing damage I did to `std::unique_copy` ~forever ago.